gossipsub.go

```diff
@@ -899,18 +899,158 @@ func (gs *GossipSubRouter) sendRPC(p peer.ID, out *RPC) {
 		return
 	}
 
-	select {
-	case mch <- out:
-		gs.tracer.SendRPC(out, p)
-	default:
-		log.Infof("dropping message to peer %s: queue full", p)
-		gs.tracer.DropRPC(out, p)
+	dropRPC := func(rpc *RPC, template string, args ...interface{}) {
+		log.Warnf(template, args...)
+		gs.tracer.DropRPC(rpc, p)
 		// push control messages that need to be retried
-		ctl := out.GetControl()
+		ctl := rpc.GetControl()
 		if ctl != nil {
 			gs.pushControl(p, ctl)
 		}
 	}
+
+	// If we're below the max message size, go ahead and send
+	if out.Size() < gs.p.maxMessageSize {
+		select {
+		case mch <- out:
+			gs.tracer.SendRPC(out, p)
+		default:
+			dropRPC(out, "dropping message to peer %s: queue full", p)
+		}
```

Comment:
let's make a function for this, call it `GossipSubRouter.doSendRPC`, as the code is duplicated below.